spec/lang/step/intrinsics.md

```diff
@@ -383,3 +383,69 @@ impl<M: Memory> Machine<M> {
     }
 }
 ```
+
+We also implement the atomic fetch operations. First we define a helper function to decide which operations can be in a fetch operation.
+
+```rust
+/// Predicate to indicate if integer bin-op can be used for atomic fetch operations.
+fn is_atomic_binop(op: BinOpInt) -> bool {
+    use BinOpInt as B;
+    match op {
+        B::Add | B::Sub => true,
+        _ => false
+    }
+}
+
+impl<M: Memory> Machine<M> {
+    fn eval_intrinsic(
+        &mut self,
+        Intrinsic::AtomicFetch(op): Intrinsic,
```

Comment:
In terms of naming, I think this should be called `AtomicFetchAndOp`, since it doesn't just fetch.